cerebrium/container-images/defining-container-images.mdx

```diff
@@ -0,0 +1,288 @@
+
+### Getting Started
+
+The fastest and simplest way to create a config file is to run the **`cerebrium init`** command. This command creates a **`cerebrium.toml`** file in the project root, which can then be edited to suit specific application requirements.
+
+Check out the [Introductory Guide](/cerebrium/getting-started/introduction) for more information on how to get started.
+
+<Info>
+  It is possible to initialize an existing project by adding a `cerebrium.toml`
+  file to the root of your codebase, defining your entrypoint (`main.py` if
+  using the default runtime, or adding an entrypoint to the .toml file if using
+  a custom runtime) and including the necessary files in the `deployment`
+  section of your `cerebrium.toml` file.
+</Info>
+
+## Hardware Configuration
+
+Cerebrium provides flexible hardware options to match app requirements. The basic configuration specifies GPU type and memory allocations:
+
+```toml
+[cerebrium.hardware]
+compute = "AMPERE_A10" # GPU selection
+memory = 16.0          # Memory allocation in GB
+cpu = 4                # Number of CPU cores
+gpu_count = 1          # Number of GPUs
+```
+
+For detailed hardware specifications and performance characteristics see the [GPU and Other Resources Guide](#).
+
+## Dependency management
+
+### Selecting a Python Version
+
+The Python runtime version forms the foundation of every Cerebrium app. We currently support versions 3.10, 3.11 and 3.12. Specify the Python version in the deployment section of the configuration:
+
+```toml
+[cerebrium.deployment]
+python_version = 3.11
+```
+
+The Python version affects the entire dependency chain. For instance, some packages may not support newer Python versions immediately after release.
+
+<Warning>
+  Changes to the Python version trigger a full rebuild since they affect both
+  the base environment and all Python package installations.
+</Warning>
+
+### Adding Python Packages
+
+Python dependencies can be managed directly in TOML or through requirement files. The system caches packages at the node level to speed up builds:
+
+```toml
+[cerebrium.dependencies.pip]
+torch = "==2.0.0"
+transformers = "==4.30.0"
+numpy = "latest"
+```
+
+Or using an existing requirements file:
+
+```toml
+[cerebrium.dependencies.paths]
+pip = "requirements.txt"
+```
+
+<Tip>
+  For GitHub repositories, use shell commands instead of pip dependencies to
+  ensure proper versioning.
+</Tip>
+
+The system implements an intelligent caching strategy at the node level. When an app is built, all pip packages are cached with their exact versions, including wheel files and compiled binaries. This means subsequent builds only need to install new or updated packages, significantly reducing build times.
```

Comment:
same about "at the node level"

We cache each of the build steps in layers. Each step is a layer that can be cached. Any change to an earlier layer requires all later layers to be rebuilt - same as docker